test/core/simd/simd_lane.wast

```diff
@@ -393,36 +393,67 @@
   (v128.const i64x2 01_234_567_890_123_456_789_0 0x0_1234_5678_90AB_cdef))
   (v128.const i32x4 0xeb1f_0ad2 0xab54_a98c 0x90ab_cdef 0x1234_5678))
 
+;; Malformed lane index value
+
+(assert_malformed (module (func (result i32) (i8x16.extract_lane_s  -1 (v128.const i8x16 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0)))) "malformed lane index")
```

Comment:
I may be misinterpreting the intention, but I was under the impression that `assert_malformed` should largely always be used with `(module quote "...")` because in theory this sort of parse error doesn't work in that when we try to parse the `assert_malformed` directive it will fail since the contents are designed to not parse.

Reply:
WAVM is more permissive here than it probably should be: an error in an unquoted module that the parser can recover from without skipping out of the `assert_malformed` will be treated the same as if the module was quoted.

Reply:
Searching around https://github.com/webassembly/testsuite it looks like all `assert_malformed` directives are either coupled with `(module quote` or `(module binary` currently.

Reply:
@alexcrichton @Honry https://github.com/WAVM/WAVM/commit/165863c4542aaecd93c9de4d84d3fae523a9036e makes it an error to use anything other than a quoted or binary module in `assert_malformed`.

Reply:
Nice! That looks great to me, thanks! 👍

Reply:
Thank you @AndrewScheidecker, @alexcrichton, I will make a copy of the fix from WAVM to this PR.

Reply:
Done!